### ISSUE TITLE
For Megatron models, local vocab files are not supported.

### ISSUE THREAD

**Issue by meetdave06** [NVIDIA/NeMo#1604]
https://github.com/NVIDIA/NeMo/blob/51092ef7a9810ab096148cdf2d44ddfe6c72d028/nemo/collections/nlp/modules/common/megatron/megatron_utils.py#L135

It seems from the above code snippet that for Megatron models, it always tries to download the vocab files from the cloud. However, currently for bio-Megatron models the [link](https://api.ngc.nvidia.com/v2/models/nvidia/biomegatron345mcased/versions/0/files/vocab.txt) doesn't work. This has been already raised as a separate [issue](https://github.com/NVIDIA/NeMo/issues/1582). 

It should accept the vocab file passed to `config.model.tokenizer.vocab_file` and should only check for vocab file online, if the user doesn't provide a vocab file.

ERROR while running Token Classification - BioMegatron notebook

<img width="1080" alt="Screen Shot 2021-01-05 at 8 24 51 PM" src="https://user-images.githubusercontent.com/7240751/103717828-14580900-4f94-11eb-925b-25ce3650b002.png">

**Comment by ericharper** [comment]
I was able to run 
```
vocab = get_megatron_vocab_file('biomegatron-bert-345m-uncased')
```
without any issues.

I was wondering if you could help clarify what you're trying to do.

Are you looking to use 'biomegatron-bert-345m-uncased', or are you pretraining a megatron language model on your own vocab and data? Or are you wanting to use 'biomegatron-bert-345m-uncased' configuration but pretrained on your own vocab and data?

Also, there is a known bug with the notebook and config interpolation which we are working on: https://github.com/NVIDIA/NeMo/issues/1582

**Comment by meetdave06** [comment]
I am just trying to run the notebook to understand NeMo framework better.

Initially, I tried to just pass "biomegatron-bert-345m-uncased" to `config.model.language_model.pretrained_model_name = PRETRAINED_BERT_MODEL` and it didn't work as mentioned here #1582 .

So I downloaded the model, config, and vocab file for Megatron models from the links given in this notebook. However, even if I passed the vocab file manually, the line

`vocab = get_megatron_vocab_file('biomegatron-bert-345m-uncased') 
`

doesn't check for it and always tries to download it from the cloud given the pre-trained model name. I hope this makes it clear.
